### PR TITLE
Implement ban history

### LINKED ISSUE #140: Use sane date format for specifying ban duration
Right now it uses `%m/%d/%Y %I:%M %p`, which shouldn't even be a thing.  Use ISO 8601 date format instead.

Maybe also try out how `Time.parse` from `require "time"` parses certain strings -- perhaps we don't need a specific format then.

https://github.com/Retrospring/retrospring/blob/5ae5ed6c8dd6f1b150f19543574a63069f46313a/app/controllers/ajax/moderation_controller.rb#L118

### PR DESCRIPTION
**Pull request by raccube**
Closes #140 

- [x] Tests
- [x] Update Rake tasks
- [x] Update Rails Admin
- [x] Frontend: Unban

### CODE REVIEW

**Review by pixeldesu — APPROVED** [review]
LGTM 👍

File: app/models/user.rb [hunk 1]
```diff
@@ -218,21 +223,31 @@ def report_comment(report, content)
...
   def unban
-    self.update(permanently_banned: false, ban_reason: nil, banned_until: nil)
+    UseCase::User::Unban.call(id)
   end
 
-  def ban(buntil=nil, reason=nil)
-    if buntil == nil
-      self.update(permanently_banned: true, ban_reason: reason)
+  # Bans a user.
+  # @param duration [Fixnum, nil] Ban duration
```
Comment (nilsding) [inline app/models/user.rb]: Fixnum doesn't really exist these days, `Integer` is where it's at
```suggestion
  # @param duration [Integer?] Ban duration
```

File: spec/models/user_ban_spec.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe UserBan, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (nilsding) [inline spec/models/user_ban_spec.rb]: Delete this

**Review by nilsding — CHANGES_REQUESTED** [review]

File: spec/models/user_ban_spec.rb (hunk 2 above)
Comment (raccube) [reply spec/models/user_ban_spec.rb]: 669aa579

File: app/models/user.rb (hunk 1 above)
Comment (raccube) [reply app/models/user.rb]: 669aa579

**Review by nilsding — APPROVED** [review]

**Pull request merged**